### PR TITLE
plugins/catppuccin!: update integrations

### PR DESCRIPTION
**Pull request by mrtnvgr**
This PR updates catppuccin integrations and sorts them to alphabetical order

### CODE REVIEW

File: plugins/colorschemes/catppuccin.nix [hunk 1]
```diff
@@ -199,6 +273,10 @@ in {
 
         noice = helpers.defaultNullOpts.mkBool config.plugins.noice.enable "";
 
+        # TODO: NormalNvim = ;
```
Comment (mrtnvgr) [inline plugins/colorschemes/catppuccin.nix:276]: How can I rename this option to snake_case?

**Review by mrtnvgr — COMMENTED** [review]
Should all integrations relay on their plugin enabled state?

**Comment by mrtnvgr** [comment]
I think we need a type which would allow for `integrations.mini = enable` and `integrations.mini.enabled = true`. Catppuccin [does this at compile time](https://github.com/catppuccin/nvim/blob/ea52fe8a0b1e4a820df0d0cf9a6a5a0e18c3eaa0/lua/catppuccin/lib/mapper.lua#L29-L39).

File: plugins/colorschemes/catppuccin.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/colorschemes/catppuccin.nix:302]: `normal_nvim` ?

Comment (mrtnvgr) [reply plugins/colorschemes/catppuccin.nix:302]: > `normal_nvim` ?

This integration's name is in camel case, will it work? https://github.com/catppuccin/nvim/commit/0e3c128eea8a7de692778d52b8429817df5c9040

Comment (GaetanLepage) [reply plugins/colorschemes/catppuccin.nix:302]: Oh, I didn't understand well. I would say that we leave it as `NormalNvim` (same as upstream).

**Comment by GaetanLepage** [comment]
@mrtnvgr is your PR now ready for review ? If so, don't hesitate to undraft it accordingly so that I take a closer look.

File: plugins/colorschemes/catppuccin.nix [hunk 2]
```diff
@@ -144,21 +144,52 @@ in {
...
+          bold_basename = helpers.defaultNullOpts.mkBool true "";
+
+          dim_context = helpers.defaultNullOpts.mkBool false "";
+
+          alt_background = helpers.defaultNullOpts.mkBool false "";
+        };
+
         beacon = helpers.defaultNullOpts.mkBool false "";
 
+        # TODO: bufferline = ;
```
Comment (GaetanLepage) [inline plugins/colorschemes/catppuccin.nix:163]: You still have a few TODOs left

Comment (mrtnvgr) [reply plugins/colorschemes/catppuccin.nix]: These TODOs are special integrations which require additional setup in other plugins. I need help to implement them

Comment (GaetanLepage) [reply plugins/colorschemes/catppuccin.nix]: Ok, let's leave them as is for now.

**Review by GaetanLepage — APPROVED** [review]

**Pull request merged**